### DISCUSSION TITLE
[Question]: RAG without uploading a file.

### DISCUSSION THREAD

**Discussion by shota1998** [danny-avila/LibreChat#2894] [Q&A] [answered]
### What is your question?

Does RAG only function when a file is attached to the message?
Is there a setting that allows RAG to always function even without attaching a file?

### More Details

I have my own vector data in db. 
I want to reference this from LibreChat using the RAG API.

I have checked the LibreChat documentation but couldn't find the relevant information.

I have already confirmed that the rag works in the message with a file attached.

### What is the main subject of your question?

Documentation, Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
> Does RAG only function when a file is attached to the message?

Yes

> Is there a setting that allows RAG to always function even without attaching a file?

In a future update, there is something planned for this.

**Reply by lusob** [reply]
> > Does RAG only function when a file is attached to the message?
> 
> Yes
> 
> > Is there a setting that allows RAG to always function even without attaching a file?
> 
> In a future update, there is something planned for this.

Hi @danny-avila ! De we have some news about that 👆 ?

Thanks!

**Reply by Umair444** [reply]
I think it is still not done. Even with assistants, I always have to upload the file to make the RAG from RAG_API working.

**Reply by wongtatseng** [reply]
Hmmm…I was looking at the same problem, trying to connect to our own vector db and couldn’t find a way.
